### PR TITLE
Add Cypress e2e testing

### LINKED ISSUE #1038: UI Behavioural Testing
Integrate some limited UI testing to Kavita

https://testing-angular.com/

### PR DESCRIPTION
**Pull request by TheIceCreamTroll**
# Added
- Added: End-to-end testing using Cypress (Closes #1038)

To run a test from the Cypress GUI, run `ng e2e`, then go to `Start E2E Testing > Specs` and run each spec in order

To run all specs from the CLI, run `ng run kavita-webui:cypress-run --browser firefox`. Edge and Chrome are also supported if they are installed on your system. If you don't have Firefox installed, alter the command to a different browser. The config will need to be cleared between each CLI run for now.

### CODE REVIEW

**Comment by majora2007** [comment]
First off, please update the Contributing.md to include how to run Cypress. Second, let me open a branch for you to target into, so that we can all work on it and @therobbiedavis can help us setup the CI/CD.

I think the easiest way to approach the testing library is to utilize a sub repo of public domain works. This would allow us to keep the main repo light and only for cypress testing would the subrepo need to be pulled.

**Comment by majora2007** [comment]
Are you fine if I merge this and we can work off the new branch together?

**Comment by TheIceCreamTroll** [comment]
> Are you fine if I merge this and we can work off the new branch together?

Ready to merge

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,18 +44,26 @@ Setup guides, FAQ, the more information we have on the [wiki](https://wiki.kavit
...
+### Testing ###
+We use [Cypress](https://www.cypress.io/) for end-to-end testing before each release
+- Make sure the API is running in Visual Studio/Rider before continuing
+- To open the Cypress GUI, run `ng e2e`, then go to `Start E2E Testing > Specs`. The specs can be found in `Kavita\UI\Web\cypress\e2e`
+- To run all specs in order from the command line, run `ng run kavita-webui:cypress-run --browser firefox`
+  - If you don't have Firefox installed, substitute `--browser firefox` with `--browser edge` or `--browser chrome`
+- To run a specific spec from the command line, run `ng run kavita-webui:cypress-run --browser firefox --spec "cypress\e2e\SPEC_FILE"`
+
 ### Swagger API ###
 If you just want to play with Swagger, you can just
```
Comment (ThePromidius) [inline CONTRIBUTING.md:64]: Maybe add that it can be enabled from the Kavita instance without the need to build the whole Kavita?

**Pull request merged**